src/utils/catchAsync.util.ts

```diff
@@ -0,0 +1,16 @@
+import { IRequest, IResponse, INextFunction } from '../interfaces/vendor';
+import ApiResponse from './ApiResponse';
+
+interface IController {
+    (req: IRequest, res: IResponse): Promise<ApiResponse>;
+}
```

Comment:
doesn't IController need a next function in it ?

Reply:
I don't think so, by default it forwards the request to next middleware

Reply:
- NextFunction will be required for Middlewares only not Controllers.
- As of know, I don't see any scenarios where controllers need NextFunction.

Reply:
got it